lib/orbit/transformable.js

```diff
@@ -68,50 +71,34 @@ var Transformable = {
     if (object._transformable === undefined) {
       object._transformable = true;
       object._transformationQueue = new ActionQueue();
+      object._activeTransformations = [];
 
       Evented.extend(object);
 
-      object.didTransform = function(operation, inverse) {
-        var normalized = normalize(operation);
-        var transformation = transformationFor.call(this, normalized);
-        if (transformation) {
-          // console.log('Transformable#didTransform - matching transformation found', this.id, normalized, inverse);
-          transformation.pushCompletedOperation(normalized, inverse);
-
-        } else {
-          // console.log('Transformable#didTransform - createTransformation', this.id, normalized, inverse);
-          transformation = new Transformation(this);
-          transformation.pushCompletedOperation(normalized, inverse);
-          queueTransformation.call(this, transformation);
-        }
+      object.didTransform = function() {
+        Orbit.deprecate('`didTransform` has been deprecated. Please call `transformed` instead and supply a TransformResult as an argument.', true);
       };
 
-      object.currentTransformation = function() {
-        if (this._transformationQueue.current) return this._transformationQueue.current.data;
+      object.transformed = function(result) {
+        return this.applyTransformation( new Transformation(null, result) );
```

Comment:
> Somethings feels a bit off here. Passing a TransformationResult into a Transformation seems back to front and in combination with the null for operations just looks wrong (will need to keep digging).

A `Transformation` can represent both the applied operations and the result of applying those operations. I originally had separate `Transform` and `TransformResult` classes, so that `didTransform` emitted both. The ancestry info was kept on the `Transform`. The advantage of combining them into a single `Transformation` was easier queueing and the ability to emit everything together. I did struggle with the modeling of these concepts for quite a while and am open to suggestions.